src/inmanta/data/__init__.py

```diff
@@ -4827,6 +5003,131 @@ def done(self) -> int:
             return self.total
         return self._done
 
+    @classmethod
+    async def create_and_insert(
+        cls,
+        env_id: uuid.UUID,
+        version: int,
+        total: int,
+        version_info: Optional[JsonType],
+        undeployable: abc.Sequence[ResourceIdStr],
+        skipped_for_undeployable: abc.Sequence[ResourceIdStr],
+        partial_base: Optional[int] = None,
+        rids_in_partial_compile: Optional[abc.Set[ResourceIdStr]] = None,
+        connection: Optional[Connection] = None,
+    ) -> "ConfigurationModel":
+        """
+        Create a new ConfigurationModel instance and insert it into the database. When partial_base is provided,
+        the new ConfigururationModel will contain all the undeployables and skipped_for_undeployables present
+        in the partial_base version that are not part of the partial compile, i.e. not present in rids_in_partial_compile.
+        """
+        rids_in_partial_compile = rids_in_partial_compile if rids_in_partial_compile is not None else set()
+        now = datetime.datetime.now().astimezone()
+        if partial_base is None:
+            cm = ConfigurationModel(
+                environment=env_id,
+                version=version,
+                date=now,
+                total=total,
+                version_info=version_info,
+                undeployable=undeployable,
+                skipped_for_undeployable=skipped_for_undeployable,
+            )
+            await cm.insert(connection=connection)
+        else:
+            # A partial compile was done. Create the new version of the ConfigurationModel based on the partial_base version.
+            query = f"""
+                WITH rids_undeployable_base_version AS (
```

Comment:
is this correct? 

This works if we assume there are no cross-set-dependencies, but do we enforce that?

Reply:
I added this check.